code/datums/components/deadchat_control.dm

```diff
@@ -0,0 +1,240 @@
+	input_cooldown = _input_cooldown
+	on_removal = _on_removal
+	if(deadchat_mode & DEADCHAT_DEMOCRACY_MODE)
+		if(deadchat_mode & DEADCHAT_ANARCHY_MODE) // Choose one, please.
+			stack_trace("deadchat_control component added to [parent.type] with both democracy and anarchy modes enabled.")
+		timerid = addtimer(CALLBACK(src, PROC_REF(democracy_loop)), input_cooldown, TIMER_STOPPABLE | TIMER_LOOP)
+
+	var/list/input_names = list()
+	for(var/item in inputs)
+		input_names |= item
+	notify_ghosts("[parent] is now deadchat controllable! Possible commands are: [english_list(input_names)]", source = parent, action = NOTIFY_FOLLOW, title="Something Interesting!")
+	if(!ismob(parent) && !(parent in GLOB.poi_list))
+		GLOB.poi_list |= parent
+		generated_point_of_interest = TRUE
+	message_admins("[parent] has been given deadchat control in [deadchat_mode == DEADCHAT_ANARCHY_MODE ? "anarchy" : "democracy"] mode with a cooldown of [input_cooldown SECONDS] second\s.")
+
+/datum/component/deadchat_control/Destroy(force, silent)
+	var/message = "<span class='deadsay italics bold'>[parent] is no longer controllable.</span>"
+	for(var/mob/dead/observer/M in orbiters)
+		to_chat(M, message)
+	on_removal?.Invoke()
+	inputs = null
+	orbiters = null
+	ckey_to_cooldown = null
+	if(generated_point_of_interest)
+		GLOB.poi_list -= parent
+	return ..()
+
+/datum/component/deadchat_control/proc/deadchat_react(mob/source, message)
+	SIGNAL_HANDLER
+
+	message = lowertext(message)
+
+	if(!inputs[message])
+		return
+
+	if(deadchat_mode & DEADCHAT_ANARCHY_MODE)
+		if(!source || !source.ckey)
+			return
+		var/cooldown = ckey_to_cooldown[source.ckey] - world.time
+		if(cooldown > 0)
+			to_chat(source, "<span class='warning'>Your deadchat control inputs are still on cooldown for another [CEILING(cooldown * 0.1, 1)] second\s.</span>")
+			return MOB_DEADSAY_SIGNAL_INTERCEPT
+		ckey_to_cooldown[source.ckey] = world.time + input_cooldown
+		addtimer(CALLBACK(src, PROC_REF(end_cooldown), source.ckey), input_cooldown)
+		inputs[message].Invoke()
+		to_chat(source, "<span class='notice'>\"[message]\" input accepted. You are now on cooldown for [input_cooldown * 0.1] second\s.</span>")
+		return MOB_DEADSAY_SIGNAL_INTERCEPT
+
+	if(deadchat_mode & DEADCHAT_DEMOCRACY_MODE)
+		ckey_to_cooldown[source.ckey] = message
+		to_chat(source, "<span class='notice'>You have voted for \"[message]\".</span>")
+		return MOB_DEADSAY_SIGNAL_INTERCEPT
+
+/datum/component/deadchat_control/proc/democracy_loop()
+	if(QDELETED(parent) || !(deadchat_mode & DEADCHAT_DEMOCRACY_MODE))
+		deltimer(timerid)
+		return
+	var/result = count_democracy_votes()
+	if(!isnull(result))
+		inputs[result].Invoke()
+		if(!(deadchat_mode & MUTE_DEADCHAT_DEMOCRACY_MESSAGES))
+			var/message = "<span class='deadsay italics bold'>[parent] has done action [result]!<br>New vote started. It will end in [input_cooldown * 0.1] second\s.</span>"
+			for(var/mob/dead/observer/M in orbiters)
+				to_chat(M, message)
```

Comment:
will this send message to people with dsay disabled? If so, it needs to be able to be filtered due to players possibly not wanting their textbox spammed with a low-delay dchat plays ss13 game.

Reply:
It should only send to people who are directly orbiting the object. That being said, I can definitely account for that on all these to_chats.

Reply:
Addressed, added some checks to account for deadchat mutes.